node/pkg/common/obsvReqSendC.go

```diff
@@ -0,0 +1,19 @@
+package common
+
+import (
+	"fmt"
+
+	gossipv1 "github.com/certusone/wormhole/node/pkg/proto/gossip/v1"
+)
+
+const ObsvReqChannelSize = 50
+const ObsvReqChannelFullError = "channel is full"
+
+func PostObservationRequest(obsvReqSendC chan *gossipv1.ObservationRequest, req *gossipv1.ObservationRequest) error {
+	if len(obsvReqSendC) >= cap(obsvReqSendC) {
+		return fmt.Errorf(ObsvReqChannelFullError)
+	}
+
+	obsvReqSendC <- req
+	return nil
```

Comment:
Since this was already merged I don't think we need to change it but I believe the idiomatic way to do this in go is to use a select statement:

```suggestion
select {
	obsvReqSendC <- req:
		return nil
	default:
		return ObsvReqChannelFullError
}
```

Since any goroutine can push to a channel the channel could become full immediately after the length check, which would cause the send to block.  `select` avoids this issue because if the actual send blocks then it will fall through to the default statement.

I think the part of guardiand that calls this method is single-threaded so it doesn't matter in practice but we need to make sure we update this if we ever start pushing into this channel from multiple goroutines.